\documentclass{article}

%% almost as usual
\author{Ludwig A. Hothorn$^1$ and Torsten Hothorn$^2$\\
$^1$Institute of Biostatistics, Leibniz University of Hannover \\
Herrenh{\"a}user Stra{\ss}e 2, D--30419 Hannover, Germany\\
$^2$Institute for Medical Informatics, Biometry and Epidemiology\\
Friedrich--Alexander--Universit\"at Erlangen--N\"urnberg \\
Erlangen, Germany}

\title{Order-restricted scores test for the evaluation of population-based 
       case-control studies when the genetic model is unknown}
%% TH: kopier Dir bitte das Sweave.sty in Deinen tex-Pfad oder einfach in 
%% Verzeichnis, in dem Du Tex rennen laesst.
\usepackage{Sweave}
\usepackage[round]{natbib}

\usepackage{verbatim}
\usepackage{graphicx}
\usepackage{amsfonts}
\usepackage{amstext}
\usepackage{amsmath}
%%\usepackage{amsthm}
%%\usepackage[round]{natbib}
%%\usepackage{bibentry}
%%\usepackage{hyperref}
\usepackage{thumbpdf}
\usepackage{rotating}
%%\usepackage{floatflt}

\newcommand{\R}{\mathbb{R} }
\newcommand{\Prob}{\mathbb{P} }
\newcommand{\N}{\mathbb{N} }
\newcommand{\C}{\mathbb{C} }
\newcommand{\V}{\mathbb{V}} %% cal{\mbox{\textnormal{Var}}} }
\newcommand{\E}{\mathbb{E}} %%mathcal{\mbox{\textnormal{E}}} }
\newcommand{\Var}{\mathbb{V}} %%mathcal{\mbox{\textnormal{Var}}} }
\newcommand{\argmin}{\operatorname{argmin}\displaylimits}
\newcommand{\argmax}{\operatorname{argmax}\displaylimits}
\newcommand{\LS}{\mathcal{L}_n}
\newcommand{\TS}{\mathcal{T}_n}
\newcommand{\LSc}{\mathcal{L}_{\text{comb},n}}
\newcommand{\LSbc}{\mathcal{L}^*_{\text{comb},n}}
\newcommand{\F}{\mathcal{F}}
\newcommand{\A}{\mathcal{A}}
\newcommand{\yn}{y_{\text{new}}}
\newcommand{\z}{\mathbf{z}}
\newcommand{\X}{\mathbf{X}}
\newcommand{\Y}{\mathbf{Y}}
\newcommand{\sX}{\mathcal{X}}
\newcommand{\sY}{\mathcal{Y}}
\newcommand{\T}{\mathbf{T}}
\newcommand{\x}{\mathbf{x}}
\renewcommand{\a}{\mathbf{a}}
\newcommand{\xn}{\mathbf{x}_{\text{new}}}
\newcommand{\y}{\mathbf{y}}
\newcommand{\w}{\mathbf{w}}
\newcommand{\ws}{\mathbf{w}_\cdot}
\renewcommand{\t}{\mathbf{t}}
\newcommand{\M}{\mathbf{M}}
\renewcommand{\vec}{\text{vec}}
\newcommand{\B}{\mathbf{B}}
\newcommand{\K}{\mathbf{K}}
\newcommand{\W}{\mathbf{W}}
\newcommand{\D}{\mathbf{D}}
\newcommand{\I}{\mathbf{I}}
\newcommand{\bS}{\mathbf{S}}
\newcommand{\cellx}{\pi_n[\x]}
\newcommand{\partn}{\pi_n(\mathcal{L}_n)}
\newcommand{\err}{\text{Err}}
\newcommand{\ea}{\widehat{\text{Err}}^{(a)}}
\newcommand{\ecv}{\widehat{\text{Err}}^{(cv1)}}
\newcommand{\ecvten}{\widehat{\text{Err}}^{(cv10)}}
\newcommand{\eone}{\widehat{\text{Err}}^{(1)}}
\newcommand{\eplus}{\widehat{\text{Err}}^{(.632+)}}
\newcommand{\eoob}{\widehat{\text{Err}}^{(oob)}}

\hyphenation{Qua-dra-tic}

\newcommand{\Rpackage}[1]{{\normalfont\fontseries{b}\selectfont #1}}
\newcommand{\Robject}[1]{\texttt{#1}}
\newcommand{\Rclass}[1]{\textit{#1}}
\newcommand{\Rcmd}[1]{\texttt{#1}}
\newcommand{\Roperator}[1]{\texttt{#1}}
\newcommand{\Rarg}[1]{\texttt{#1}}
\newcommand{\Rlevel}[1]{\texttt{#1}}

\newcommand{\RR}{\textsf{R}}
\renewcommand{\S}{\textsf{S}}

\SweaveOpts{eps=FALSE,keep.source=TRUE,echo=FALSE}

%% end of declarations %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

<<coin-setup, echo = FALSE, results = hide>>=
library("coin")
set.seed(290875)
@

\begin{document}

\maketitle

\paragraph{Key Words:}
Candidate gene association, Cochran-Armitage trend test, Maximum test, Asymptotical distribution, 
Conditional inference.

%% TH: die wollen die Einteilung in Objectives, Methods, Results, Conclusions
\begin{abstract}
	The linear trend test for proportions according to Cochran-Armitage is commonly used for genotype-based analysis of candidate gene association. Depending on the underlying genetic model, the use of model-specific scores maximizes the power. Commonly, the underlying genetic model is a priori unknown. Maximum tests using standardized Cochran-Armitage tests with the specified scores were proposed. We re-formulate the problem into a maximum test of scores-specific linear-by-linear association tests. For both maximum-type and quadratic form of test statistics the asymptotic expectation and covariance can be derived in a closed form. 
	The maximum-type test for three scores, sensitve for the additive, dominant and recessive model, is three-variate normal distributed.	A simple software using the R-package coin is presented.
	   
\end{abstract}


\section{Introduction}

In population-based case-control studies the association between a candidate allele 
and a disease can be evaluated by the Cochran-Armitage trend test \citep[CA-test][]{Armitage:1955} 
whether or not Hardy-Weinberg equilibrium holds according to \cite{Sasieni:1997}. 
The CA-test is based on a set of scores assigned to the alleles. For genotypes
$(aa, Aa, AA)$, with $A$ denoting a high risk candidate allele and $a$ any of the
other alleles, the different scores are optimal for the additive, recessive and dominant
model. Thus, if the underlying mode of inheritance is known, the choice of
the test is obvious. In situations when the underlying genetic model is unknown,
\cite{Freidlin:2002} showed that choosing the wrong test, i.e., the wrong scores, 
leads to a substancial loss of power. Consequently, inference procedures
with good power under all three genetic models are of special interest.
An intuitive idea is to use the maximum of the standardized test statistics
of the CA-tests which are optimal of the additive, recessive and dominant
model. This test was proposed and investigated by \cite{Freidlin:2002}. 
The authors derived the distribution under the null hypothesis by 
simulation because the asymptotic distribution of the so-called MAX 
test is not known.

In this paper, we show that the MAX test as suggested by \cite{Freidlin:2002} 
can be viewed as a special case of a more general conditional independence test 
\citep{StrasserWeber1999}. The asymptotic distribution of the
MAX test is derived and it is shown how the MAX test can be performed
without additional programming using existing software solutions.

\section{Maximum Test}

For case-control studies on candidate gene association the
data are typically given by the empirical 
genotype distribution in both groups.
For a simple bi-allelic marker the data can be presented 
in a $3 \times 2$ contingency table, where $A$ is the high risk candidate 
allele and $a$ is any of the other alleles (see Table~\ref{gdistr}). 

\begin{table}
\centering
\caption{Genotype distribution. \label{gdistr}}
  \begin{tabular}{l l l l}
    \hline
       & Cases & Controls & Total \\
 \hline
 aa		& 		$r_0$ & $s_0$	& $n_0$\\
 aA		& 		$r_1$ & $s_1$	& $n_1$ \\
 AA		& 		$r_2$ & $s_2$	& $n_2$ \\
 Total& 		$R$ 	& $S$		& $N$  \\
  \hline
 \end{tabular}
 \end{table}


Because in case-control studies random samples from cases and control of size $R$ and $S$ are drawn,
$r_0$,$r_1$,$r_2$ and $s_0$,$s_1$,$s_2$ follow a trinomial distribution for the genotypes
$aa$, $aA$ and $AA$ with probabilities $p_0, p_1, p_2$ and $q_0, q_1, q_2$ respectively. 
Therefore, we are interested in comparing two trinomial distributions against ordered
alternatives:
\begin{eqnarray*}
H_0: p_j = q_j \text{ vs. } H_1: p_j \prec q_j \quad \text{for } j \in \{0,1,2\}.
\end{eqnarray*}

The CA-test statistic with scores $\xi = (\xi_0, \xi_1, \xi_2)$ basically (modulo standardization) 
simply reads
\begin{eqnarray*}
\text{CA}_\xi = \xi_0 \times r_0 + \xi_1 \times r_1 + \xi_2 \times r_2.
\end{eqnarray*}
For the dominant model the scores $\xi_\text{dom}=(0,1,1)$, 
for the additive model the scores  $\xi_\text{add}=(0,1,2)$ and for the recessive model the 
scores $\xi_\text{rec}=(0,0,1)$ are power optimal \citep{Sasieni:1997,Slager:2001}. 
However, a priori the underlying genetic model is rarely known. Therefore, \cite{Freidlin:2002} 
proposed the maximum of three standardized CA-tests (called MAX test), each with 
model-specific scores, as a global test for association. In the following,
we embed the MAX test into a more general framework for conditional inference,
derive its asymptotic distribution and give an example how to practically
perform such an analysis.

\subsection{Reformulation of the problem}
Let $\Y_i$ denote the both groups (cases and controls) and $\X_i$ the genotype
for all cells $i = 1, \dots, n = 6$. The weights $w_i$ represent the number of observations in each cell.
The so-called \textit{influence function} $h$ provides us with a zero-one dummy coding of the groups (being one for
cases and zero for controls).
Moreover, three transformations $g$ of the genotype are under test: $g_\text{dom}$ 
assigns scores $\xi_\text{dom}$ to
genotypes $(aa, Aa, AA)$, $g_\text{add}$ assigns scores $\xi_\text{add}$ and $g_\text{rec}$
implements scores $\xi_\text{rec}$. See Table~\ref{gd}.

\begin{table}
\begin{center}
\begin{tabular}{lllc|cccc}
$i$ & $\Y_i$  &  $\X_i$  & $w_i$ & $h(\Y_i)$ & $g_\text{add}(\X_i)$ & $g_\text{dom}(\X_i)$ & $g_\text{rec}(\X_i)$ \\ \hline
1 & Case & aa & $r_1$ & 1 & 0 & 0 & 0 \\
2 & Case & Aa & $r_2$ & 1 & 1 & 1 & 0 \\
3 & Case & AA & $r_3$ & 1 & 2 & 1 & 1 \\
4 & Control & aa & $s_0$ & 0 & 0 & 0 & 0 \\
5 & Control & Aa & $s_1$ & 0 & 1 & 1 & 0 \\
6 & Control & AA & $s_2$ & 0 & 2 & 1 & 1 \\
\end{tabular}
\caption{Genotype distribution reformulated. \label{gd}}
\end{center}
\end{table}

\subsection{Inference Problem and Linear Statistic}

We are interested in testing the null hypothesis of independence of grouping $\Y$ and genotype $\X$
\begin{eqnarray*}
H_0: D(\Y | \X) = D(\Y)
\end{eqnarray*}
against ordered alternatives. First, we define a three-dimensional
statistic $\T$, each dimension being associated with one of the 
scores $g_\text{add}$, $g_\text{dom}$, and $g_\text{rec}$.
Each statistic is basically the sum of the scores multiplied by the weights over cases only:
\begin{eqnarray} \label{linstat}
\T = (\T_\text{add}, \T_\text{dom}, \T_\text{rec}) = 
\sum_{i = 1}^n w_i g(\X_i) h(\Y_i) \in \R^{3} 
\end{eqnarray}
with $g(\X_i) = (g_\text{add}(\X_i), g_\text{dom}(\X_i), g_\text{rec}(\X_i))$.

\subsection{Conditional Expectation and Covariance}

The distribution of $\T$  depends on the joint
distribution of $\Y$ and $\X$, which is unknown under almost all practical
circumstances. At least under the null hypothesis one can dispose of this
dependency by fixing the genotypes and conditioning on all possible
permutations of the groups. 
This principle leads to test procedures known as \textit{permutation tests}. 
\cite{StrasserWeber1999} derived closed-form expressions for the conditional 
expectation $\mu \in \R^{pq}$ and covariance $\Sigma \in \R^{3 \times 3}$ 
of $\T$ under $H_0$ given all permutations of the groupings.

Let $w_\cdot = \sum_{i = 1}^n w_i$ denote the sum of the weights.
The conditional expectation of the influence function $h$ is
\begin{eqnarray*}
\E(h) = w_\cdot^{-1} \sum_i w_i h(\Y_i) \in \R
\end{eqnarray*}
with corresponding variance
\begin{eqnarray*}
\V(h) = w_\cdot^{-1} \sum_i w_i \left(h(\Y_i) - \E(h)\right)^2
\end{eqnarray*}
The conditional expectation of the linear statistic $\T$ is 
\begin{eqnarray}
\mu & = & \E(\T) = \E(h) \sum_{i = 1}^n w_i g(\X_i), \nonumber \\
\Sigma & = & \V(\T) \nonumber \\
& = &
    \frac{w_\cdot}{w_\cdot - 1}  \V(h) \times
        \left(\sum_i w_i \left( g(\X_i) g(\X_i)^\top\right) \right)
\label{expectcovar}
\\
& - & \frac{1}{w_\cdot - 1}  \V(h)  \times \left(
        \sum_i  w_i g(\X_i) \right) \left( \sum_i w_i g(\X_i)\right)^\top.
\nonumber
\end{eqnarray}

Note that the complete covariance structure, and thus the correlation between the elements
of the three-dimensional statistic $\T$ is known and can be computed for the data at hand.
The corresponding correlation matrix coincides with the correlations
obtained for the three CA-test statistic by \cite{Freidlin:2002}.

\subsection{Test Statistics}

Based on the three-dimensional statistic $\T$ and its expectation 
$\mu$ and covariance matrix $\Sigma$,
we can easily construct test statistics and derive their distribution under the conditions
described in the null hypothesis. As the sum of the weights $w_\cdot$ tends to infinity, 
\cite{StrasserWeber1999}
proved that the limiting distribution of the three-dimensional statistic $\T$ is a three-dimensional
normal distribution with mean $\mu$ and covariance $\Sigma$. Thus, the asymptotic distribution
of a maximum-type statistic
\begin{eqnarray*}
c_\text{max}(\T, \mu, \Sigma)  = \max \left| \frac{\T - \mu}{\text{diag}(\Sigma)^{1/2}} \right|
\end{eqnarray*}
can be evaluated by computing three-dimensional normal probabilities. Alternatively, a quadratic
form 
\begin{eqnarray*}
c_\text{quad}(\T, \mu, \Sigma)  = (\T - \mu)^\top \Sigma^+ (\T - \mu)
\end{eqnarray*}
follows a $\chi^2_2$ distribution. The quadratic form, which is a competitor for the MAX test statistic,
reveals high power for an average alternative 
while the maximum-type form for a particular genetic alternative. Therefore, we focus on the 
maximum-type statistics, particularly because information on the elementary genetic alternative 
is available by multiplicity-adjusted $p$-values. Note that, under any circumstances, the exact conditional
distribution can be approximated by conditional Monte-Carlo methods, which is especially attractive
for small sample sizes $w_\cdot$ when we can't expect asymptotics to work well.

\section{Illustration}

In order to compare the above test and its implementation with the
results reported by \cite{Freidlin:2002}, we reanalyse a study on association
between a variant of the epidermal growth factor 
(EGF) gene and malignant melanoma according \citep[Table~\ref{mel},][]{Shahbazi:2002}.

<<melanoma-table, echo = FALSE>>=
me <- as.table(matrix(c( 6,  8, 10,
               32, 47, 20), byrow = TRUE, nrow = 2,
    dimnames = list(group = c("In situ", "Control"),
                    genotype = c("AA", "AG", "GG"))))
me <- t(me)
@

\begin{table}[h]
\begin{center}
\caption{Melanoma data. \label{mel}}
\begin{tabular}{l l l l l} \hline
<<melanoma-tabular, echo = FALSE, results = tex>>=
metab <- cbind(me, Total = rowSums(me))
metab <- rbind(metab, Total = colSums(metab))
cat(" & ", paste(colnames(metab), collapse = " & "), "\\\\ \\hline \n")
tmp <- sapply(1:nrow(metab), function(i)
    cat(rownames(metab)[i], " & ", paste(metab[i,], collapse = " & "), 
        " \\\\ \n"))
@
\hline
\end{tabular}
\end{center}
\end{table}
                
In \RR \citep{rcore2007} the data are represented as table object
<<melanoma-data, echo = TRUE>>=
me
@
We only have to implement the function $g$ assigning scores
$\xi_\text{add}$, $\xi_\text{dom}$ and $\xi_\text{rec}$ to each
observation based on its genotype, i.e., a matrix with three columns
<<melanoma-setup, echo = TRUE>>=
add <- c(0, 1, 2)
dom <- c(0, 1, 1)
rec <- c(0, 0, 1)
g <- function(x) {
    x <- unlist(x)
    cbind(add = add[x], dom = dom[x], rec = rec[x])
}
@
The function \Rcmd{independence\_test} from the \Rpackage{coin} \citep{PKG:coin}
can now be used to perform the test as described above:
<<melanoma-test, echo = TRUE>>=
it <- independence_test(me, xtrafo = g, alternative = "greater")
@
The statistic $\T$ is
<<melanoma-stat, echo = TRUE>>=
statistic(it, "linear")
@
We immediately are provided with the covariance matrix
$\Sigma$ and corresponding correlation matrix
<<melanoma-cor, echo = TRUE>>=
cov2cor(covariance(it))
@
those values coincide with the correlations between the 
three different CA-test statistics given by \cite{Freidlin:2002}.
The MAX test with test statistic $c_\text{max}$ and its 
asymptotic $p$-value is
<<melanoma-tp, echo = TRUE>>=
statistic(it)
pvalue(it)
@
which is the same $p$-value as shown in Table~8 of \cite{Freidlin:2002}.
This global $p$-value does not give any information about the
underlying genetic model. Multiplicity-adjusted $p$-values
for each of the additive, dominant and recessive tests, indicate
which model is likely (see Section~\ref{sim} in addition):
<<melanoma-adj, echo = TRUE>>=
pvalue(it, method = "single-step")
@
It seems that the recessive model descibes the mode of inheritance best
in this case.
We might want to check whether the asymptotics work well enough
in this situation. The exact conditional
$p$-value is approximated by a conditional Monte-Carlo procedure
with $49999$ random permutations of the data:
<<melanoma-resampl, echo = TRUE>>=
itp <- independence_test(me, xtrafo = g,
    distribution = approximate(B = 49999), 
    alternative = "greater")
@
The $p$-value is
<<melanoma-itpval, echo = TRUE>>=
pvalue(itp)
@
and the step-down multiplicity-adjusted $p$-values \citep{WestfallYoung1993}
can be computed via
<<melanoma-sd, echo = TRUE>>=
pvalue(itp, method = "step-down")
@
We refer to \cite{Hothorn:2006:AmStat} for more examples.

\section{A simulation study \label{sim}}
Many different patterns of penetrances $f_i$, disease prevalence $p$, sample size of cases and controls $R$, $S$ can be investigated in a simulation study. We will focus on a high prevalent disease (i.e. $p=0.5$), penetrances according to a pure additive, recessive and dominant genetic model (as well as no association) for a total sample size of $N=400$ divided into the balanced $R=S=200$ and several unbalanced sampling schemes. Unbalanced data are of interest because real data examples exist with seriously more controls, see e.g. the data in Table 1, or with seriously more cases, see e.g. the IL13 polymorphism in atopic dermatitis [10, Table 4]. For the proposed asymptotic order-restricted test (ASM-test) both the global power $\pi_{global}$(the decision rate in favor of any alternative) and the correct model identification rates $\nu_{add}$,$\nu_{rec}$,$\nu_{dom}$ are compared with the power of the individual genetic model tests $\pi_{add}$, $\pi_{rec}$, $\pi_{dom}$ in Table 4.

\begin{table}
\centering
\caption{Type I error rate and empirical power estimates: prevalence $p=0.5$, 10000 runs}
  \begin{tabular}{l l l l l l l l l l}
    \hline
        Model & R & S & $\pi_{global}$ &  $\nu_{add}$ & $\nu_{rec}$ & $\nu_{dom}$ & $\pi_{add}$ & $\pi_{rec}$ & $\pi_{dom}$   \\
 \hline
 Null & 200 & 200 & 0.048&	0.012 &	0.017& 	0.019 			&	0.051 &	0.047 &	0.049 \\
 Dom  & 200 & 200 & 0.85 &	0.13 &	0.01 &	\textit{0.71} &	0.75 &	0.23 &	\textbf{0.91} \\
 Add  & 200 & 200 & 0.84 &	\textit{0.53} &	0.10  &	0.21 &	\textbf{0.88} &	0.71 &	0.78 \\
 Rec  & 200 & 200 & 0.86 &	0.16 &	\textit{0.69} &	0.01 & 	0.80 &	\textbf{0.91} &	0.28 \\
 Dom  & 100 & 300 & 0.72 &	0.16 &	0.01 &	\textit{0.55}&	0.63 &	0.21 & \textbf{0.80} \\
 Add  & 100 & 300 & 0.73 &	\textit{0.43} &	0.14 &	0.16 &	\textbf{0.78} &	0.60 &	0.65 \\
 Rec  & 100 & 300 & 0.77 &	0.16 &	\textit{0.60} &	0.01 &	0.67 &	\textbf{0.82} &	0.22  \\
 Dom  & 300 & 100 & 0.76 &	0.11 &	0.01 &	\textit{0.64} &	0.66 &	0.19 &	\textbf{0.82} \\
 Add  & 300 & 100 & 0.75 &	\textit{0.42} &	0.09 &	0.24 &	\textbf{0.79} &	0.59 &	0.69 \\
 Rec  & 300 & 100 & 0.75 &	0.17 &	\textit{0.55} &	0.01 &	0.69 &	\textbf{0.82 }&	0.24 \\
\hline
 \end{tabular}
 \end{table} 

 
Per definition all tests control the type I error rates. Clearly, the power is maximal for the individual, unadjusted tests when the genetic model is known (bold). But the a priori knowledge of the genetic model is unrealistic. For balanced samples sizes the power of the order restricted scores test (ORS-test) is independent of the underlying genetic model and non-inferior smaller compared with the maximum power for the known model. Additionally to the global decision that a significant association exists, the ORS-test provide an adjusted p-value for the most likely genetic model. Clearly, these model identification rates are seriously smaller then the global power. Hereby is the identification of the additive model most difficult because of the both equal competitors, where the identification of the dominant or recessive model is easier (and equal) because only the additive model is the competitor. For unbalanced designs the power decreases although the total sample size remains constant.

\section{Discussion}
After re-formulation of Sasieni's problem, a three-variate normal distributed maximum-type 
order-restricted scores test was proposed. The decision for or against the global 
hypothesis of association can be performed by a global p-value. If the global 
null hypothesis can be rejected, the likeliness of a selected genetic alternatives, 
i.e. the dominant or additive or recessive model, can be stated by means of 
multiplicity-adjusted p-values. Simulations show that the loss of the global 
power of the new approach compared with that assuming a particular genetic 
model is acceptable. After a significant global test is the identification of 
either the dominant or the recessive model feasible. The identification of an 
additive model is more difficult because of two competing alternatives. 
A simple software solution using the R -package coin is available.


\bibliographystyle{plainnat}
\bibliography{references}

\end{document}


